addons/project/models/project.py

```diff
@@ -1846,7 +1846,7 @@ def default_get(self, default_fields):
             if project.analytic_account_id:
                 vals['analytic_account_id'] = project.analytic_account_id.id
         else:
-            vals['user_ids'] = [Command.link(self.env.user.id)]
+            vals['user_ids'] = self._context.get('default_user_ids', False) or [Command.link(self.env.user.id)]
```

Comment:
Isn't a `Command.link` needed too?
Because if `default_get` is called from a write, this will override the task's current user_ids rather than add. Right?

Reply:
@auon-odoo , indeed. Updated the code. Thanks

Reply:
Isn't `vals['user_ids'] = [Command.link(self._context.get('default_user_ids', False) or self.env.user.id)]` not enough?

Reply:
@auon-odoo , We already have default values inside the `vals['user_ids']` so it should be better not to replace it because we are creating the private task(without project) so just add the current user as an assignee and kept the rest to the assignee as it is. 
Thanks

Reply:
Ok. Thanks for you explanation :)